codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -265,18 +271,21 @@ public Http2Stream resetSent() {
         }
 
         @Override
-        public final Object setProperty(Object key, Object value) {
-            return data.put(key, value);
+        public final <K, V> V setProperty(PropertyKey<K, V> key, V value) {
+            checkNotNull(key, "key");
+            return properties.add((DefaultPropertyKey<K, V>) key, value);
```

Comment:
@Scottmitch 
keep in mind that if we support `PropertyKey` independent of a connection, we are 1) also independent of an `EventLoop` and 2) such a "static" key has to be able to be created at any time and 3) we have to support channel re-registration between `EventLoop`s.